projects/plugins/jetpack/_inc/lib/admin-pages/class.jetpack-react-page.php

```diff
@@ -271,13 +271,18 @@ public function page_admin_scripts() {
 		if ( file_exists( $script_deps_path ) ) {
 			$asset_manifest      = include $script_deps_path;
 			$script_dependencies = $asset_manifest['dependencies'];
+			$asset_version       = $asset_manifest['version'];
 		}
 
+		$version = ( defined( 'WP_DEBUG' ) && WP_DEBUG && isset( $asset_version ) )
+			? $asset_version
+			: JETPACK__VERSION;
```

Comment:
Probably better to do like
```suggestion
		$version = JETPACK__VERSION;
		if ( file_exists( $script_deps_path ) ) {
			$asset_manifest      = include $script_deps_path;
			$script_dependencies = $asset_manifest['dependencies'];
			$version             = $asset_manifest['version'];
		}
```
I can't see any reason not to use the asset version if it's available. Actually, I'm unclear as to when the built script would exist _without_ the corresponding asset.php file.

I note we also reference this same script in class.jetpack-connection-banner.php, perhaps that should use the same versioning logic. And the corresponding CSS file is for some reason registered in class.jetpack.php.

IMO the best solution would be to switch the script over to using `Assets::register_script()`, then figuring out how to properly reconcile the three places referencing its parts.

Reply:
> IMO the best solution would be to switch the script over to using Assets::register_script()

I like that option; that would clean things up indeed.

> then figuring out how to properly reconcile the three places referencing its parts.

I'm not sure what you mean by that? Are you referring to the 3 inline scripts below this? If we don't change the script slug, those should keep working I think. They seem to in my tests. I pushed 6b598ccf081abae6a158a11f44c8151c80e49274 to give this a try.

Reply:
I was referring to
https://github.com/Automattic/jetpack/blob/68ca0434f056da0320366a79ffd53b9eeba91bcc/projects/plugins/jetpack/class.jetpack-connection-banner.php#L259
and
https://github.com/Automattic/jetpack/blob/68ca0434f056da0320366a79ffd53b9eeba91bcc/projects/plugins/jetpack/class.jetpack.php#L4095-L4100

Do we care that we might wind up loading the CSS twice now, and maybe the JS too?

Reply:
Ah, I missed that, it's true this one ends up being a bit trickier since it can be used on the plugins page. I've reverted and switched to your first suggestion above in a922ba189b875ddd95d07690cc043b43586f565b